source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,620 @@
+
+/*-----------------------------------------------------------*/
+
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_rx_bd_struct_t rxBuffDescrip_0[ ENET_RXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_tx_bd_struct_t txBuffDescrip_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static rx_buffer_t rxDataBuff_0[ ENET_RXBUFF_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static tx_buffer_t txDataBuff_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+
+static void prvEMACHandlerTask( void * pvParameters );
+
+static void ethernet_callback( ENET_Type *base,
+                               enet_handle_t *handle,
+                               enet_event_t event,
+                               enet_frame_info_t *frameInfo,
+                               void *userData );
+
+static void prvProcessFrame( int length );
+
+BaseType_t xNetworkInterfaceInitialise( void )
+{
+	status_t xStatus;
+	phy_config_t xConfig = { 0 };
+	BaseType_t xResult = pdPASS;
+	phy_speed_t speed;
+    phy_duplex_t duplex;
+    status_t readStatus;
+    enet_config_t config;
+    uint32_t sysClock;
+    enet_buffer_config_t buffCfg[ENET_RING_NUM];
+    int i;
+    uint8_t ucCounter = 0;
+    bool autoNegotiationComplete = false;
+    bool LinkUp = false;
+
+    configASSERT( FSL_FEATURE_ENET_QUEUE == 1 );
+
+    #if defined( ENET_ENHANCEDBUFFERDESCRIPTOR_MODE )
+        configASSERT( pdFALSE );
+	#endif
+
+    ethernetifLocal->RxBuffDescrip = &( rxBuffDescrip_0[ 0 ] );
+    ethernetifLocal->TxBuffDescrip = &( txBuffDescrip_0[ 0 ] );
+    ethernetifLocal->RxDataBuff    = &( rxDataBuff_0[ 0 ] );
+    ethernetifLocal->TxDataBuff    = &( txDataBuff_0[ 0 ] );
+    ethernetifLocal->base          = ( void * ) ENET_BASE;
+
+	/* Set the clock frequency. */
+	mdioHandle.resource.csrClock_Hz = EXAMPLE_CLOCK_FREQ;
+	mdioHandle.resource.base = ( void * ) ENET_BASE;
+
+	/* Set the configuration to auto-negotiate; set the phy to full duplex mode; the phy's address; and the speed. */
+	xConfig.autoNeg = pdTRUE;
+	xConfig.duplex = kPHY_FullDuplex;
+	xConfig.phyAddr = BOARD_ENET0_PHY_ADDRESS;
+	xConfig.speed = kPHY_Speed100M;
+
+	configPRINTF( ( "Starting PHY initialization.\r\n" ) );
+
+	xStatus = PHY_Init( &phyHandle, &xConfig );
+
+	if( xStatus != kStatus_Success )
+	{
+		xResult = pdFAIL;
+		configPRINTF( ( "Failed to initialize the PHY..." ) );
+	}
+	else
+	{
+		configPRINTF( ( "Configuration successful. Waiting for autonegotiation to complete...\r\n" ) );
+
+		do
+		{
+			readStatus = PHY_GetAutoNegotiationStatus( &phyHandle, &autoNegotiationComplete );
+
+			if( autoNegotiationComplete == true )
+			{
+				break;
+			}
+
+			/* Try for only a limited number of times. */
+			if( ucCounter++ > MAX_AUTONEG_FAILURE_COUNT )
+			{
+				break;
+				xResult = pdFAIL;
+			}
+
+			vTaskDelay( pdMS_TO_TICKS( 10 ) );
+		}
+		while( readStatus == kStatus_Success );
+	}
+
+
+	if( xResult == pdPASS )
+	{
+		if( autoNegotiationComplete == false )
```

Comment:
Were you able to this this scenario? Auto-negotiation fails and a manual configuration succeeds?

Reply:
I removed this. Because this would force a default speed if the cable is disconnected and reconnected at just the right moment.